### PR TITLE
fix: scope must be a ref

### PR DESCRIPTION
**Pull request by RJWadley**
this is universally better

the old type forced you to break the rules of react and allowed you to do things that didn't make sense

### CODE REVIEW

File: useAnimation.ts [hunk 1]
```diff
@@ -33,7 +33,7 @@ const useAnimation = <F, T>(
 	createAnimations: F extends VoidFunction ? F : () => T,
 	deps: DependencyList,
 	options?: {
-		scope?: string | Element | null
+		scope?: React.RefObject<HTMLElement | null>
```
Comment (RJWadley) [inline useAnimation.ts:36]: test

File: useAnimation.ts [hunk 2]
```diff
@@ -100,7 +100,7 @@ const useAnimation = <F, T>(
 			} else {
 				setReturnValue(undefined)
 			}
-		}, options?.scope ?? undefined)
+		}, options?.scope?.current ?? undefined)
```
Comment (RJWadley) [inline useAnimation.ts:103]: test2

File: useAnimation.ts [hunk 3]
```diff
@@ -33,7 +33,7 @@ const useAnimation = <F, T>(
 	createAnimations: F extends VoidFunction ? F : () => T,
 	deps: DependencyList,
 	options?: {
-		scope?: string | Element | null
+		scope?: React.RefObject<HTMLElement | null>
 		kill?: boolean
```
Comment (davidthecomposer) [inline useAnimation.ts:37]: only refs huh?

**Review by davidthecomposer — APPROVED** [review]
Looks good

File: useAnimation.ts (hunk 2 above)
Comment (RJWadley) [inline useAnimation.ts:103]: test

**Pull request merged**